docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,1459 @@
+
+In this topology, the FRR instance on each node establishes an eBGP session with
+its upstream router. The upstream routers are then responsible for advertising
+the node subnet routes to the rest of the network. It is also possible for
+different groups of nodes (e.g. nodes in different racks) to belong to different
+Autonomous Systems.
+
+```mermaid
+graph LR
+    subgraph AS 65002 [AS 65002: Rack 1]
+        direction LR
+        NodeA(Node A)
+        NodeB(Node B)
+    end
+    subgraph AS 65003 [AS 65003: Rack 2]
+        direction LR
+        NodeC(Node C)
+    end
+    subgraph AS 65001 [AS 65001: External Network]
+        direction LR
+        ToR1(ToR Switch 1)
+        ToR2(ToR Switch 2)
+    end
+    NodeA <-- eBGP --> ToR1
+    NodeB <-- eBGP --> ToR1
+    NodeC <-- eBGP --> ToR2
+    ToR1  <-- iBGP --> ToR2
+```
+
+##### Full-Mesh iBGP across the Cluster
+
+The FRR instance on each node maintains a full-mesh BGP peer relationship with
+all other nodes across the cluster. In this mode, users can enable no-overlay
+mode without relying on external BGP routes. Enabling `outboundSNAT` for the
+no-overlay network allows pod egress traffic to be SNATed to the node IP,
+enabling communication with external destinations even when the pod network is
+not advertised externally.
+
+This is particularly useful in smaller clusters or in environments where
+external BGP speakers are not available. However, it is important to note that a
+full-mesh iBGP setup can lead to increased CPU and memory consumption on the
+nodes, as each node must maintain a session with every other node.
+
+```mermaid
+graph LR
+    subgraph AS 64512 [ AS 64512 ]
+      direction LR
+      A(Node A)
+      B(Node B)
+      C(Node C)
+      A <-- iBGP --> B
+      B <-- iBGP --> C
+      C <-- iBGP --> A
+    end
+```
+
+Here's an example FRRConfiguration:
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  labels:
+    network: default
+  name: receive-all
+  namespace: frr-k8s-system
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      # All nodes are added as neighbors to setup a full-mesh BGP topology
+      neighbors:
+      - address: 192.168.111.20
+        asn: 64512
+        disableMP: true
+        toReceive:
+          allowed:
+            mode: all
+      - address: 192.168.111.21
+        asn: 64512
+        disableMP: true
+        toReceive:
+          allowed:
+            mode: all
+      - address: 192.168.111.22
+        asn: 64512
+        disableMP: true
+        toReceive:
+          allowed:
+            mode: all
+```
+
+##### External Route Reflectors for Larger Clusters
+
+In a large cluster, a full-mesh BGP setup leads to more CPU and memory
+consumption on the nodes. Instead of every node peering with every other node,
+nodes peer only with external BGP route reflectors. This significantly reduces
+the number of BGP sessions each node needs to maintain, improving scalability.
```

Comment:
what does this have to do with no overlay ? I get why this is useful, and why ovn-k should want to implement it.
I just don't see what's the relationship of that with no overlay mode. 

@pliurh